tests/test_openalex.py

```diff
@@ -0,0 +1,50 @@
+from urllib.parse import urlparse, parse_qs
+
+import pytest
+from paper_feeds.config import config
+from paper_feeds.services.openalex import (
+    _openalex_get,
+    get_journal_homepages
+)
+
+
+def test_most_basic_api_call():
+    """
+    Test bare call to the OpenAlex API (no endpoint or params specified).
+    """
+    response = _openalex_get('', {})
+    assert response.status_code == 200
+
+    # Assert header for 'User agent' was included
+    user_agent_header = response.request.headers.get('User-Agent', None)
+    assert 'paper-feeds' in user_agent_header
+
+    # Assert email was not set (default in config is None)
+    params = parse_qs(urlparse(response.url).query)
+    assert 'mailto' not in params
+
+
+def test_email_included_in_params(mocker):
+    """
+    Test setting of 'mailto' in query parameters if parameter exists in config.
+    """
+    # set config parameter using mock
+    email = 'tests@example.com'
+    mocker.patch.object(config, "crossref_email", new=email)
+
+    # Assert email was set to value in config
+    response = _openalex_get('', {})
+    params = parse_qs(urlparse(response.url).query)
+    assert params['mailto'][0] == email
+
+
+@pytest.mark.parametrize(
+    'issn,homepage',
+    [
+        ('0024-2160', 'https://academic.oup.com/library'),
+        ('1758-6909', None)
+    ]
+)
+def test_get_journal_homepages(issn, homepage):
+    issn_homepage_map = get_journal_homepages([issn])
+    assert issn_homepage_map[issn] == homepage
```

Comment:
one additional test would be great here just to ensure that the `update_journals_homepage_urls` function works and correctly writes to the db :). I think that would just be something like

```python
@pytest.mark.parametrize(
    'issn,homepage',
    ...
)
def test_update_journal_homepages(issn, homepage):
    update_journals_homepage_urls([issn])
    journal = load_journal(issn)
    assert journal.homepage_url == homepage

```

Reply:
separated business logic from db operations (`journal_service` vs `journal_repository`) and added tests for both methods in `journal_repository`